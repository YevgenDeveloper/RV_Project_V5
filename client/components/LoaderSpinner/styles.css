@import 'client/styles/variables';

.loader {
  z-index: 100;
  margin: auto;
  position: relative;
  display: block;
}

.loaderSpinner {
  position: absolute;
  top: 50%;
  left: 50%;
  margin: -25px 0 0 -25px;
  width: 50px;
  height: 50px;

  &::before {
    content: '';
    display: block;
    padding-top: 100%;
  }
}

.loaderCircular {
  position: absolute;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  transform-origin: center center;
  animation: loaderCircular 1.8s linear infinite;
}

.loaderPath {
  stroke-dasharray: 1, 200;
  stroke-dashoffset: 0;
  stroke-linecap: round;
  stroke: $color-blue;
  animation: loaderPath 0.9s ease-in-out infinite;
}

@keyframes loaderCircular {
  100% {
    transform: rotate(360deg);
  }
}

@keyframes loaderPath {
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -35;
  }

  100% {
    stroke-dasharray: 89, 200;
    stroke-dashoffset: -124;
  }
}
