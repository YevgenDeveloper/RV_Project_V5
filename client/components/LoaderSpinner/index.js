// @flow

import React from 'react';
import classNames from 'classnames/bind';
import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  className?: string,
};

const LoaderSpinner = (props: Props) => (
  <div className={cx(styles.loader, props.className)}>
    <div className={styles.loaderSpinner}>
      <svg className={styles.loaderCircular} viewBox="25 25 50 50">
        <circle
          className={styles.loaderPath}
          cx="50"
          cy="50"
          r="20"
          fill="none"
          strokeWidth="3"
          strokeMiterlimit="10"
        />
      </svg>
    </div>
  </div>
);

LoaderSpinner.defaultProps = {
  className: undefined,
};

export default LoaderSpinner;
