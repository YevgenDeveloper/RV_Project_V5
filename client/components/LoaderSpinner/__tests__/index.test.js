// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import LoaderSpinner from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<LoaderSpinner />);

  expect(wrapper).toMatchSnapshot();
});
