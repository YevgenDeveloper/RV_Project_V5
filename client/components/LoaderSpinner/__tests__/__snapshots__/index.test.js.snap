// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoaderSpinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="loaderSpinner"
      >
        <svg
          className="loaderCircular"
          viewBox="25 25 50 50"
        >
          <circle
            className="loaderPath"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />
        </svg>
      </div>,
      "className": "loader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          className="loaderCircular"
          viewBox="25 25 50 50"
        >
          <circle
            className="loaderPath"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />
        </svg>,
        "className": "loaderSpinner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <circle
            className="loaderPath"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="3"
          />,
          "className": "loaderCircular",
          "viewBox": "25 25 50 50",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "loaderPath",
            "cx": "50",
            "cy": "50",
            "fill": "none",
            "r": "20",
            "strokeMiterlimit": "10",
            "strokeWidth": "3",
          },
          "ref": null,
          "rendered": null,
          "type": "circle",
        },
        "type": "svg",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="loaderSpinner"
        >
          <svg
            className="loaderCircular"
            viewBox="25 25 50 50"
          >
            <circle
              className="loaderPath"
              cx="50"
              cy="50"
              fill="none"
              r="20"
              strokeMiterlimit="10"
              strokeWidth="3"
            />
          </svg>
        </div>,
        "className": "loader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="loaderCircular"
            viewBox="25 25 50 50"
          >
            <circle
              className="loaderPath"
              cx="50"
              cy="50"
              fill="none"
              r="20"
              strokeMiterlimit="10"
              strokeWidth="3"
            />
          </svg>,
          "className": "loaderSpinner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <circle
              className="loaderPath"
              cx="50"
              cy="50"
              fill="none"
              r="20"
              strokeMiterlimit="10"
              strokeWidth="3"
            />,
            "className": "loaderCircular",
            "viewBox": "25 25 50 50",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "loaderPath",
              "cx": "50",
              "cy": "50",
              "fill": "none",
              "r": "20",
              "strokeMiterlimit": "10",
              "strokeWidth": "3",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          "type": "svg",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
